flocx_market/db/sqlalchemy/api.py

```diff
@@ -60,9 +60,13 @@ def offer_get_all_by_project_id(context):
         project_id=context.project_id).all()
 
 
-def offer_get_all_by_server_id(context, server_id, status=None):
+def offer_get_all_by_resource(context,
```

Comment:
I'm not sure I understand why both `resource_id` and `resource_type` are required here. Shouldn't `resource_id` be unique even across resource types? It doesn't seem like we would need to know the resource type here.

Reply:
Generally: shouldn't resource type be an attribute of the resource, rather than of Offers or other objects?

Reply:
*Is* resource_id guaranteed to be unique? Are we sure we'd only ever consider OpenStack that use a uuid as their primary identifier?